src/dex/jarvis-v6/jarvis-v6.ts

```diff
@@ -0,0 +1,473 @@
+          dexKey,
+          network,
+          dexHelper,
+          this.logger,
+          pool,
+          this.priceFeed,
+          this.poolInterface,
+        )),
+    );
+  }
+
+  // Initialize pricing is called once in the start of
+  // pricing service. It is intended to setup the integration
+  // for pricing requests. It is optional for a DEX to
+  // implement this function
+  async initializePricing(blockNumber: number) {
+    const poolStates = await getOnChainState(
+      this.dexHelper,
+      this.priceFeed.address,
+      this.poolConfigs,
+      this.poolInterface,
+      blockNumber,
+    );
+
+    this.poolConfigs.forEach((pool, index) => {
+      const eventPool = this.eventPools[pool.address.toLowerCase()];
+      eventPool.setState(poolStates[index], blockNumber);
+      this.dexHelper.blockManager.subscribeToLogs(
+        eventPool,
+        eventPool.addressesSubscribed,
+        blockNumber,
+      );
+    });
+  }
+
+  // Returns the list of contract adapters (name and index)
+  // for a buy/sell. Return null if there are no adapters.
+  getAdapters(side: SwapSide): { name: string; index: number }[] | null {
+    return this.adapters[side];
+  }
+
+  getEventPool(srcToken: Token, destToken: Token): JarvisV6EventPool | null {
+    const pool = getJarvisPoolFromTokens(srcToken, destToken, this.poolConfigs);
+    return this.eventPools[pool?.address.toLowerCase()!];
+  }
+
+  // Returns list of pool identifiers that can be used
+  // for a given swap. poolIdentifiers must be unique
+  // across DEXes. It is recommended to use
+  // ${dexKey}_${poolAddress} as a poolIdentifier
+  async getPoolIdentifiers(
+    srcToken: Token,
+    destToken: Token,
+    side: SwapSide,
+    blockNumber: number,
+  ): Promise<string[]> {
+    const eventPool = this.getEventPool(srcToken, destToken);
+    if (!eventPool) return [];
+    return [eventPool.getIdentifier()];
+  }
+
+  async getPoolState(
+    pool: JarvisV6EventPool,
+    blockNumber: number,
+  ): Promise<PoolState> {
+    const eventState = pool.getState(blockNumber);
+    if (eventState) return eventState;
+    const onChainState = await pool.generateState(blockNumber);
+    pool.setState(onChainState, blockNumber);
+    return onChainState;
+  }
+
+  // Returns pool prices for amounts.
+  // If limitPools is defined only pools in limitPools
+  // should be used. If limitPools is undefined then
+  // any pools can be used.
+  // Pair
+  async getPricesVolume(
+    srcToken: Token,
+    destToken: Token,
+    amounts: bigint[],
+    side: SwapSide,
+    blockNumber: number,
+    limitPools?: string[],
+  ): Promise<null | ExchangePrices<JarvisV6Data>> {
+    const eventPool = this.getEventPool(srcToken, destToken);
+
+    if (!eventPool) return null;
+    const poolAddress = eventPool.poolConfig.address.toLowerCase();
+    const poolIdentifier = eventPool.getIdentifier();
+    if (limitPools && !limitPools.includes(poolIdentifier)) return null;
```

Comment:
This line should be commented out with a note that since this DEX is constant price and we have fixed routing (via USDC) which will cause problems for route aggregation (multiple routes with jToken -> USDC), we should not use pool allocations.

Reply:
done buy this was not enough, had to keep my change on api side